astropy/stats/tests/test_funcs.py

```diff
@@ -440,6 +442,42 @@ def test_mad_std():
         data = normal(5, 2, size=(100, 100))
         assert_allclose(funcs.mad_std(data), 2.0, rtol=0.05)
 
+def test_mad_std_scalar_return():
+    with NumpyRNGContext(12345):
+        data = normal(5, 2, size=(10, 10))
+        # make a masked array with no masked points
+        data = np.ma.masked_where(np.isnan(data), data)
+        rslt = funcs.mad_std(data)
+        # want a scalar result, NOT a masked array
+        assert np.isscalar(rslt)
+
+        data[5,5] = np.nan
+        rslt = funcs.mad_std(data, ignore_nan=True)
+        assert np.isscalar(rslt)
+        with catch_warnings() as warns:
+            rslt = funcs.mad_std(data)
+            if NUMPY_GE_1_13:
+                # for recent numpy versions, numpy masked arrays appear to
```

Comment:
So, funny thing (at least in numpy 11.11.2 and, apparently, 1.12.0b1): 

`np.ma.median(data)` does *not* return `nan` is `data` is a masked array, even if all of the mask values are `False`. It returns the ~~median ignoring the nan~~....well, no it doesn't. But it returns a number. 

If `data` is *not* masked, then `np.ma.median` *does* return `nan`.

This behavior is different in numpy 1.10, in which `np.ma.median` returns `nan` if `data` contains an unmasked `nan`.

This is clearly a numpy bug, but how should it be dealt with here? The issue is not so much the test as it is the fact that right now, in numpy 1.11, if the data contains a `nan` then the result of `mad_std` will be `nan` if data is passed in as `np.array` but will return a number if data is passed in as `np.ma.array` even if the mask is `False`.